lib/private/Avatar/AvatarManager.php

```
@@ -152,4 +171,33 @@ public function deleteUserAvatar(string $userId): void {
 	public function getGuestAvatar(string $name): IAvatar {
 		return new GuestAvatar($name, $this->logger);
 	}
+
+	public function getGenericAvatar(string $type, string $id): IAvatar {
+		if ($type === 'user') {
+			return $this->getAvatar($id);
+		}
+
+		if ($type === 'guest') {
+			return $this->getGuestAvatar($id);
+		}
+
+		$context = $this->bootstrapCoordinator->getRegistrationContext();
+
+		if ($context === null) {
+			// Too early, nothing to do yet
+			throw new \InvalidArgumentException('Unknown avatar type: ' . $type);
```

I'd argue that we are lying here. The argument isn't invalid, but the code can't handle the request. Another exception would be more appropriate IMO.

Of course this is highly unlikely to happen because we do the registration before anything else. Hence failing hard here could be an option.

---

> I'd argue that we are lying here. The argument isn't invalid, but the code can't handle the request. Another exception would be more appropriate IMO.

I agree, although I used `InvalidArgumentException` because I did not find any suitable exception and adding a new exception just for this seemed unnecessary (given that in practice it will not happen). So exception suggestions welcome :-)

> Of course this is highly unlikely to happen because we do the registration before anything else. Hence failing hard here could be an option.

Failing hard as in `exit()`? Having an exit there would feel very out of place in my opinion. Or do you mean `throw new Error()`? Or something else? :-)

---

> I agree, although I used `InvalidArgumentException` because I did not find any suitable exception and adding a new exception just for this seemed unnecessary (given that in practice it will not happen). So exception suggestions welcome :-)

In https://github.com/nextcloud/server/pull/24702/files#diff-7bfcce99ae84563547086798084d75003c7099fedeabb795d2c01962293b33fcR88 I went with `\RuntimeException`.

> Failing hard as in `exit()`? Having an exit there would feel very out of place in my opinion. Or do you mean `throw new Error()`? Or something else? :-)

I meant letting the exception bubble up, without any explicit handling :)

---

> In https://github.com/nextcloud/server/pull/24702/files#diff-7bfcce99ae84563547086798084d75003c7099fedeabb795d2c01962293b33fcR88 I went with \RuntimeException.

`\RuntimeException` it is then.

> > Failing hard as in exit()? Having an exit there would feel very out of place in my opinion. Or do you mean throw new Error()? Or something else? :-)
>
> I meant letting the exception bubble up, without any explicit handling :)

Ok, that makes a lot more sense :-P